src/realm/obj.cpp

```diff
@@ -1935,6 +1935,60 @@ bool Obj::remove_backlink(ColKey col_key, ObjLink old_link, CascadeState& state)
 }
 
 void Obj::assign(const Obj& other)
+{
+    copy_other_object(other);
+    auto copy_links = [this, &other](ColKey col) {
+        auto t = m_table->get_opposite_table(col);
+        auto c = m_table->get_opposite_column(col);
+        auto backlinks = other.get_all_backlinks(col);
+        for (auto bl : backlinks) {
+            auto linking_obj = t->get_object(bl);
+            if (c.get_type() == col_type_Link) {
+                // Single link
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == other.get_key());
+                linking_obj.set(c, get_key());
+            }
+            else {
+                auto l = linking_obj.get_linklist(c);
+                auto n = l.find_first(other.get_key());
+                REALM_ASSERT(n != realm::npos);
+                l.set(n, get_key());
+            }
+        }
+        return false;
+    };
+    m_table->for_each_backlink_column(copy_links);
+}
+
+void Obj::dup_and_handle_multiple_backlinks()
+{
+    auto copy_links = [this](ColKey col) {
+        auto t = m_table->get_opposite_table(col);
+        auto c = m_table->get_opposite_column(col);
+        auto backlinks = get_all_backlinks(col);
+
+        for (auto bl : backlinks) {
+            auto obj = m_table->create_object();
+            obj.copy_other_object(*this);
```

Comment:
This copies all outgoing links exactly. If the destination of an outgoing link is itself an embedded object then this produces multiple links to that object which is semantically incorrect. What you'll have to do is copy the entire forward link chain of embedded objects (if any). I'd recommend that you use the `InterRealmObjectConverter` that I wrote for client resets because it already does this and handles all property types that you are missing here. I think you'd just have to move it down from a sync namespace to a lower level so that you can access it here.

Reply:
this has now been handled.